### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

**Comment by allenwq** [comment]
@kxmbrian @weiqingtoh @fonglh need your comments, or you guys can directly work on my branch if there're too many parts to modify 😅

**Comment by weiqingtoh** [comment]
@allenwq updated - can check if what I edited is correct?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -52,6 +56,25 @@ track of the number of pending mail deliveries.
...
+To do unit tests on components, we use Enzyme's `shallow` rendering to render the component, then
+use jest's `expect` together with the [snapshot](https://facebook.github.io/jest/docs/snapshot-testing.html) 
+feature to assert the component's state given different inputs/props.
+
+#### Integration test
+To do integration tests, stub or mock the server side APIs, then assert whether the API receives the
+correct set of params from the tested components. All components including `redux` (actions, reducers, etc)
+and `react-router` should be tested in the integration test.
+
+Enzyme's `mount` rendering should be used in integration test. Where possible, try to avoid
```
Comment (kxmbrian) [inline CONTRIBUTING.md]: integration ~test~ tests

**Comment by weiqingtoh** [comment]
@fonglh @kxmbrian any comments?

**Comment by allenwq** [comment]
@weiqingtoh Thanks for the amendment, LGTM !

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ Code styles are predefined in the RubyMine/IntelliJ project files. Also, the
 [Rails Community Style Guide](https://github.com/bbatsov/ruby-style-guide) covers majority of the
 styles that we use.
 
+Our frontend code style follows Airbnb's [JavaScript](https://github.com/airbnb/javascript) and
+[React](https://github.com/airbnb/javascript/tree/master/react) style guide. Run `yarn lint` in the
```
Comment (fonglh) [inline CONTRIBUTING.md]: `[React's] style guide`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -52,6 +56,25 @@ track of the number of pending mail deliveries.
 
 Trackable Jobs can be waited -- use the `TrackableJob#wait` method.
 
+### Frontend
+`enzyme` and `jest` are used for frontend testing. All tests files should be put under a folder
```
Comment (fonglh) [inline CONTRIBUTING.md]: test~~s~~

Or

"Files for all tests"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -52,6 +56,25 @@ track of the number of pending mail deliveries.
...
+named `__test__`, which is excluded by webpack from the main bundler.
+
+#### Unit tests
+To do unit tests on components, we use Enzyme's `shallow` rendering to render the component, then
+use jest's `expect` together with the [snapshot](https://facebook.github.io/jest/docs/snapshot-testing.html) 
+feature to assert the component's state given different inputs/props.
+
+#### Integration test
```
Comment (fonglh) [inline CONTRIBUTING.md]: `Integration tests`

**Comment by fonglh** [comment]
Just some minor typos. LGTM once those are fixed!

File: CONTRIBUTING.md (hunk 2 above)
Comment (weiqingtoh) [inline CONTRIBUTING.md]: Actually original seems correct -> `Our frontend code style follows Airbnb's Javascript and React style guide`.

**Pull request merged**